<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>try~catch</title>
</head>
<body>
    <h1>
        예상되는 에러를 try{code..} 영역에 배치하고 에러가 발생하면
        catch(error){code..} 영역에서 처리하는 예외처리 프로세스
    </h1>
    <ul>
        <li>
            try 영역 : 에러 발생 가능성 있는 코드 실행 영역
        </li>
        <li>
            catch(error) : try 영역에서 발생한 에러 처리 영역
        </li>
        <li>
            try~catch 구문을 사용하면 프로세스 중지없이 지속가능
        </li>
        <li>
            finally 영역 : 예외처리 관계없이 무조건 실행되는 코드 영역
        </li>
        <li>
            throw : 예외를 고의적으로 발생시키는 키워드 throw + 예외 발생 후 처리할 코드
        </li>
    </ul>
    <script>
        console.log('프로세스 시작!');
        try{
            alertt('경고창 실행');
            console.log('경고창 실행 확인');
        }catch(err){
            console.log('catch 영역')
            console.log(err.name);
            console.log(err.message);
            try{
                if(err.message.toString().length > 0){
                    throw 'throw error';
                }
            }
            catch(err2){
                console.log('err2 '+err2.message);
            }
            finally{
                console.log('err2 finally 문장');
            }
        }finally{
            console.log('err finally 문장');
        }
        console.log('프로세스 종료!');
    </script>
</body>
</html>